cloud/lib/authentication/identityProviders/appleIdentityProvider.ts

```diff
@@ -0,0 +1,38 @@
+import { EnvName } from "@enums/EnvName";
+import { aws_cognito as cognito } from "aws-cdk-lib";
+import { CognitoUserPoolCdkConstruct } from "../cognitoUserPoolCdkConstruct";
+
+export const appleIdentityProvider = (
+  construct: CognitoUserPoolCdkConstruct,
+  envName: EnvName
+): cognito.UserPoolIdentityProviderApple => {
+  const appleClientId = process.env.IDENTITY_PROVIDER_APPLE_CLIENT_ID;
+  const appleKeyId = process.env.IDENTITY_PROVIDER_APPLE_KEY_ID;
+  const applePrivateKey = process.env.IDENTITY_PROVIDER_APPLE_PRIVATE_KEY;
+  const appleTeamId = process.env.IDENTITY_PROVIDER_APPLE_TEAM_ID;
+
+  if (appleClientId && appleKeyId && applePrivateKey && appleTeamId)
```

Comment:
We're duplicating this logic in other identityProviders (checking if variables exists, throwing specific errors). Can we make some generic functions for it?

On the other hand, please keep in mind that it's always preferred to fail fast. Maybe we could make some research, if we can check existance of all required env variables at the very start of deployment? 🤔

Reply:
`bin/environment` is now validating if all required variables exists and throws an error immediately